frontend/src/App.js

```diff
@@ -11,6 +11,41 @@ const baseURL = "http://localhost:3001"; //move to .env
 function App() {
   const [loading, setLoading] = useState(false);
   const [listOfPizzas, setListOfPizzas] = useState([]);
+  const [cartItems, setCartItems] = useState([]); //initially cart items is empty
+
+  //adding to cart event handler
+  const onAdd = (pizza) => {
+    console.log("on add");
+    const exist = cartItems.find((x) => x.pizzaid === pizza.pizzaid); //check if pizzaid is already in the list
+    //if already available, then increase quantity
+    if (exist) {
+      //only update the particular pizza's quantity (else keep as it is)
+      setCartItems(
+        cartItems.map((x) =>
+          x.pizzaid === pizza.pizzaid ? { ...exist, qty: exist.qty + 1 } : x
```

Comment:
What happens here when the pizzaID is no there? How is it related to exist?

Reply:
'exist' stores the boolean state, that verifies if the given pizzaitem is already on the cart. 

- If the given pizzaid found on the cartItems list, then cartItems's qty is increased by 1.
- If its not found on the list, then pizzaitem is added to the cart with a qty of 1

Reply:
could you have done this in a "flatter" manner. Everytime you add an if and else you diverge the code into paths that is not accessible by other paths.


You already have the object at the find position
And you are doing another loop to add the exist part.

Cant you try to reuse the state instead of having to loop and search two times?